File: BankTest/Models/Deposit.cs
namespace BankBackendApp.Models
{
    public class Deposit
    {
        public int id { get; set; }
        public int user_id { get; set; }
        public int deposit_type_id { get; set; }
        public int currency_id { get; set; }
        public decimal amount { get; set; }
        public decimal interest_rate { get; set; }
        public DateTime start_date { get; set; }
        public DateTime end_date { get; set; }
        public bool is_active { get; set; }
        public DateTime created_at { get; set; }

        public User User { get; set; }
        public DepositType DepositType { get; set; }
        public Currency Currency { get; set; }
    }

}

File: client/src/types/deposit.ts
import { Currency } from "./currency";

export interface IDepositType {
    can_add_money: boolean;
    can_close_early: false;
    description: string;
    id: number;
    interest_rate: number;
    max_months: number;
    min_months: number;
    name: string;
};

export interface IDeposit {
    id: number;
    amount: number;
    interest_rate: number;
    is_active: boolean;
    start_date: string;
    end_date: string;
    currency: Currency;
    depositType: IDepositType;
};
